CodeEdit/Features/Settings/Pages/ThemeSettings/Models/ThemeModel+CRUD.swift

```diff
@@ -0,0 +1,292 @@
+                if isImporting != true {
+                    self.themes[index].author = NSFullUserName()
+                    self.save(self.themes[index])
+                }
+                self.selectedTheme = self.themes[index]
+                self.detailsTheme = self.themes[index]
+            }
+        } catch {
+            print("Error adding theme: \(error.localizedDescription)")
+        }
+    }
+
+    func rename(to newName: String, theme: Theme) {
+        do {
+            guard let oldURL = theme.fileURL else {
+                throw NSError(
+                    domain: "ThemeModel",
+                    code: 1,
+                    userInfo: [NSLocalizedDescriptionKey: "Theme file URL not found"]
+                )
+            }
+
+            var iterator = 1
+            var finalName = newName
+            var finalURL = themesURL.appendingPathComponent(finalName).appendingPathExtension("cetheme")
+
+            // Check for existing display names in themes
+            while themes.contains(where: { theme != $0 && $0.displayName == finalName }) {
+                finalName = "\(newName) \(iterator)"
+                finalURL = themesURL.appendingPathComponent(finalName).appendingPathExtension("cetheme")
+                iterator += 1
+            }
+
+            try filemanager.moveItem(at: oldURL, to: finalURL)
+
+            try self.loadThemes()
+
+            if let index = themes.firstIndex(where: { $0.fileURL == finalURL }) {
+                themes[index].displayName = finalName
+                themes[index].fileURL = finalURL
+                themes[index].name = finalName.lowercased().replacingOccurrences(of: " ", with: "-")
+            }
+
+        } catch {
+            print("Error renaming theme: \(error.localizedDescription)")
+        }
+    }
+
+    /// Save theme to file
+    func save(_ theme: Theme) {
+        do {
+            if let fileURL = theme.fileURL {
+                let encoder = JSONEncoder()
+                encoder.outputFormatting = [.sortedKeys]
+                let data = try encoder.encode(theme)
+                let json = try JSONSerialization.jsonObject(with: data)
+                let prettyJSON = try JSONSerialization.data(withJSONObject: json, options: [.prettyPrinted])
+                try prettyJSON.write(to: fileURL, options: .atomic)
+            }
+        } catch {
+            print("Error saving theme: \(error.localizedDescription)")
+        }
+    }
+
+    /// Removes the given theme from `–/Library/Application Support/CodeEdit/themes`
+    ///
+    /// After removing the theme, themes are reloaded
+    /// from `~/Library/Application Support/CodeEdit/Themes`. See ``loadThemes()``
+    /// for more information.
+    ///
+    /// - Parameter theme: The theme to delete
+    func delete(_ theme: Theme) {
+        if let url = theme.fileURL {
+            do {
+                // remove the theme from the list
```

Comment:
I feel these 3 comments below are unnecessary since the code is readable already